standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -3466,11 +3467,49 @@ protected List<String> getJdoResult(
     }.run(true);
   }
 
-  private List<String> getPartitionNamesViaOrm(Table table, ExpressionTree tree, String order,
```

Comment:
Refactor this method because it's used in partition level, and do not needs the cd field of table, but the original parameter `Table table`  retrieved via ORM must include cd information.